File: lab 1/1.1.cpp
#include <iostream>
#include <cmath>
using namespace std;

double compare_accuracy(double a_numerator, double a_denominator, double a, double b_radicand, double b) {
    if(abs(a_numerator/a_denominator - a)/a > abs(sqrt(b_radicand)-b)/b) return b;
    return a;
}

int main() {
    double a_numerator, a_denominator, a, b_radicand, b;
    cout << "Введите числитель дроби" << endl;
    cin >> a_numerator;
    cout << "Введите знаменатель дроби" << endl;
    cin >> a_denominator;
    cout << "Введите приблеженное значение дроби" << endl;
    cin >> a;
    cout << "Введите подкоренное выражение" << endl;
    cin >> b_radicand;
    cout << "Введите приблеженное значение корня" << endl;
    cin >> b;
    cout << "Точнее " << compare_accuracy(a_numerator, a_denominator, a, b_radicand, b) << endl;
}

File: lab 1/1.1.py
import math


def compare_accuracy(a_numerator, a_denominator, a, b_radicand, b):
    """
    Возвращает более точное значение
    a_numerator - числитель
    a_denominator - знаменатель
    a - приближенное значение a
    b_radicand - подкоренное выражение
    b - приближенное значение b
    """
    return (a, b)[abs(a_numerator/a_denominator - a)/a > abs(math.sqrt(b_radicand)-b)/b]


def main():
    a_numerator, a_denominator = map(int, input("Введите числитель и знаменатель дроби через пробел ").split())
    a = float(input("Введите приблеженное значение дроби "))
    b_radicand = float(input("Введите подкоренное выражение "))
    b = float(input("Введите приблеженное значение корня "))
    print(f"Более точное приблеженное значение: {compare_accuracy(a_numerator, a_denominator, a, b_radicand, b)}")


main()
